### DISCUSSION TITLE
Library not compiling for Arduino Nano RP2040 Connect

### DISCUSSION THREAD

**Discussion by techno-wizard7** [moononournation/Arduino_GFX#487] [Q&A]
Hi, I'm trying to set the library on a Arduino Nano RP2040 Connect board, but if the library is called, I get the following errors:

`/home/helena/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_RPiPicoPAR16.cpp: In member function 'virtual bool Arduino_RPiPicoPAR16::begin(int32_t, int8_t)':
/home/helena/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_RPiPicoPAR16.cpp:14:16: error: 'digitalPinToBitMask' was not declared in this scope
   _dcPinMask = digitalPinToBitMask(_dc);
                ^~~~~~~~~~~~~~~~~~~
/home/helena/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_RPiPicoPAR16.cpp:14:16: note: suggested alternative: 'digitalPinToPinName'
   _dcPinMask = digitalPinToBitMask(_dc);
                ^~~~~~~~~~~~~~~~~~~
                digitalPinToPinName
/home/helena/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_SWPAR8.cpp: In member function 'virtual bool Arduino_SWPAR8::begin(int32_t, int8_t)':
/home/helena/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_SWPAR8.cpp:149:16: error: 'digitalPinToBitMask' was not declared in this scope
   _dcPinMask = digitalPinToBitMask(_dc);
                ^~~~~~~~~~~~~~~~~~~
/home/helena/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_SWPAR8.cpp:149:16: note: suggested alternative: 'digitalPinToPinName'
   _dcPinMask = digitalPinToBitMask(_dc);
                ^~~~~~~~~~~~~~~~~~~
                digitalPinToPinName
/home/helena/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_SWPAR16.cpp: In member function 'virtual bool Arduino_SWPAR16::begin(int32_t, int8_t)':
/home/helena/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_SWPAR16.cpp:231:16: error: 'digitalPinToBitMask' was not declared in this scope
   _dcPinMask = digitalPinToBitMask(_dc);
                ^~~~~~~~~~~~~~~~~~~
/home/helena/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_SWPAR16.cpp:231:16: note: suggested alternative: 'digitalPinToPinName'
   _dcPinMask = digitalPinToBitMask(_dc);
                ^~~~~~~~~~~~~~~~~~~
                digitalPinToPinName
/home/helena/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_RPiPicoSPI.cpp: In member function 'virtual bool Arduino_RPiPicoSPI::begin(int32_t, int8_t)':
/home/helena/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_RPiPicoSPI.cpp:32:16: error: 'digitalPinToBitMask' was not declared in this scope
   _dcPinMask = digitalPinToBitMask(_dc);
                ^~~~~~~~~~~~~~~~~~~
/home/helena/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_RPiPicoSPI.cpp:32:16: note: suggested alternative: 'digitalPinToPinName'
   _dcPinMask = digitalPinToBitMask(_dc);
                ^~~~~~~~~~~~~~~~~~~
                digitalPinToPinName
/home/helena/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_RPiPicoSPI.cpp:43:3: error: 'spi_init' was not declared in this scope
   spi_init(_spi, _spis.getClockFreq());
   ^~~~~~~~
/home/helena/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_RPiPicoSPI.cpp:43:3: note: suggested alternative: '_spi_init'
   spi_init(_spi, _spis.getClockFreq());
   ^~~~~~~~
   _spi_init
/home/helena/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_SWSPI.cpp: In member function 'virtual bool Arduino_SWSPI::begin(int32_t, int8_t)':
/home/helena/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_SWSPI.cpp:98:18: error: 'digitalPinToBitMask' was not declared in this scope
     _dcPinMask = digitalPinToBitMask(_dc);
                  ^~~~~~~~~~~~~~~~~~~
/home/helena/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_SWSPI.cpp:98:18: note: suggested alternative: 'digitalPinToPinName'
     _dcPinMask = digitalPinToBitMask(_dc);
                  ^~~~~~~~~~~~~~~~~~~
                  digitalPinToPinName
/home/helena/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_SWSPI.cpp:104:18: error: 'digitalPinToBitMask' was not declared in this scope
     _csPinMask = digitalPinToBitMask(_cs);
                  ^~~~~~~~~~~~~~~~~~~
/home/helena/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_SWSPI.cpp:104:18: note: suggested alternative: 'digitalPinToPinName'
     _csPinMask = digitalPinToBitMask(_cs);
                  ^~~~~~~~~~~~~~~~~~~
                  digitalPinToPinName
/home/helena/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_SWSPI.cpp:108:17: error: 'digitalPinToBitMask' was not declared in this scope
   _sckPinMask = digitalPinToBitMask(_sck);
                 ^~~~~~~~~~~~~~~~~~~
/home/helena/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_SWSPI.cpp:108:17: note: suggested alternative: 'digitalPinToPinName'
   _sckPinMask = digitalPinToBitMask(_sck);
                 ^~~~~~~~~~~~~~~~~~~
                 digitalPinToPinName
/home/helena/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_SWSPI.cpp:117:28: error: 'portInputRegister' was not declared in this scope
     _misoPort = (PORTreg_t)portInputRegister(digitalPinToPort(_miso));
                            ^~~~~~~~~~~~~~~~~
/home/helena/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_HWSPI.cpp: In member function 'virtual bool Arduino_HWSPI::begin(int32_t, int8_t)':
/home/helena/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_HWSPI.cpp:77:16: error: 'digitalPinToBitMask' was not declared in this scope
   _dcPinMask = digitalPinToBitMask(_dc);
                ^~~~~~~~~~~~~~~~~~~
/home/helena/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_HWSPI.cpp:77:16: note: suggested alternative: 'digitalPinToPinName'
   _dcPinMask = digitalPinToBitMask(_dc);
                ^~~~~~~~~~~~~~~~~~~
                digitalPinToPinName
/home/helena/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_RPiPicoPAR8.cpp: In member function 'virtual bool Arduino_RPiPicoPAR8::begin(int32_t, int8_t)':
/home/helena/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_RPiPicoPAR8.cpp:14:16: error: 'digitalPinToBitMask' was not declared in this scope
   _dcPinMask = digitalPinToBitMask(_dc);
                ^~~~~~~~~~~~~~~~~~~
/home/helena/Arduino/libraries/GFX_Library_for_Arduino/src/databus/Arduino_RPiPicoPAR8.cpp:14:16: note: suggested alternative: 'digitalPinToPinName'
   _dcPinMask = digitalPinToBitMask(_dc);
                ^~~~~~~~~~~~~~~~~~~
                digitalPinToPinName

exit status 1

Compilation error: exit status 1`

This is on Linux, using Arduino IDE 2.3.2, the GFX library is updated to 1.4.7, and the RP2040 board is version 4.1.3 all of which is a fresh install. I have tried getting it to work in both example code, and even calling it in an empty ino file with just calling the library without using it. Manually replacing the suggested code removes some issues, but I can't find a solution for `portInputRegister` not being declared.
is there something I am missing? Or is this library not built for the board?

**Comment by moononournation** [comment]
yes, this is not on the supported list

**Comment by PiyushXCoder** [comment]
I am getting the same issue while using Raspberry Pi Pico. Although Raspberry Pi Pico in Currently Supported Dev Board list.
